### PR TITLE
feat(routes): Capture redirects as sentry messages

### PR DESCRIPTION
**Pull request by evanpurkhiser**
This way we can see if we're actually using all of our redirects

### CODE REVIEW

**Review by scttcper — APPROVED** [review]
gonna assign these to you

**Comment by evanpurkhiser** [comment]
can always turn off if there's a lot

File: static/app/utils/reactRouter6Compat/router.tsx [hunk 1]
```diff
@@ -75,6 +76,22 @@ interface RedirectProps extends Omit<NavigateProps, 'to'> {
...
+  useEffect(() => {
+    const routePath = routes
+      .map(route => route.path ?? '')
+      .filter(path => path !== '')
+      .join('/');
+
+    Sentry.captureMessage('Redirect route used', {
+      level: 'info',
+      tags: {routePath},
+      fingerprint: [routePath],
```
Comment (markstory) [inline static/app/utils/reactRouter6Compat/router.tsx:92]: Do you want one error per route? Or have a single error with route as a tag?

Comment (evanpurkhiser) [reply static/app/utils/reactRouter6Compat/router.tsx:92]: hmm. Maybe it's OK to just have one, and we can use discover to drill down

**Review by priscilawebdev — APPROVED** [review]

**Pull request merged**